### ISSUE TITLE
Virus payload detected

### ISSUE THREAD

**Issue by collinstocks-oscar** [BashSupport-Pro/bashsupport-pro#44]
Our security team detected a malware payload in this software: https://www.virustotal.com/gui/file/8ee406486c9e98ebd2651accf43a22371d3181547ad8b33d341d0cbc47a042c0/detection

**Comment by jansorg** [comment]
@collinstocks-oscar Thanks for the notice. I'll double-check.
I've built the binaries from source on a local machine and the binaries are packed with upx (a exe compressor), perhaps that's why it was flagged. But this definitely shouldn't pop up anywhere. I'll keep you updated.

**Comment by jansorg** [comment]
(The file posted on virustotal is the binary, which is bundled with the latest version of BashSupport Pro.)

I'm pretty sure that this is a false positive. 
UPX received reports that compressed binaries are detected by antivirus software:
- https://github.com/upx/upx/issues/337
- https://github.com/upx/upx/issues/359

This binary is cross-compiled for Windows on a Linux machine, using the Golang compiler. shfmt is open-source.
The result is compressed with UPX 3.96 on the same Linux machine.

I'll look into the best way to prevent false positive detections and ship an update. I'll post here as soon as I've published the new version.
I recently submitted the binaries to a malware scan, which didn't detect any issues. But this service was using different scanners.

If you'd still like to use BashSupport Pro, just delete the .exe file, download the official builds from https://github.com/mvdan/sh/releases/tag/v3.1.2 and configure the path in the settings at `Settings > Languages & Frameworks > BashSupport Pro`.

**Comment by jansorg** [comment]
@collinstocks-oscar  Thanks for reporting, this is definitely important. It's fixed now.

I've published 1.4.14, which bundles updated binaries. These binaries are not flagged anymore on VirusTotal.

I'm certain that this wasn't a malware or a virus. UPX compression and compiler settings made TotalVirus flag the binaries (only 2 out or ~60 engines). UPX compression is turned off now. shfmt is now compiled without LDFLAG `-s` and with LDFLAG `-w` - without this change TotalVirus flagged the binary, which I built from source on a Linux machine. 
None of the updated shfmt and ShellCheck are detected as virus or malware on TotalVirus.

Changenotes and version info:
2020.2: https://plugins.jetbrains.com/plugin/13841-bashsupport-pro/versions/stable/96191
2020.1: https://plugins.jetbrains.com/plugin/13841-bashsupport-pro/versions/stable/96190
The update for 2019.3 will be published soon
